fendermint/app/src/cmd/genesis.rs

```diff
@@ -246,3 +252,64 @@ fn set_ipc_gateway(genesis_file: &PathBuf, args: &GenesisIpcGatewayArgs) -> anyh
         Ok(genesis)
     })
 }
+
+async fn new_child_genesis(
+    genesis_file: &PathBuf,
+    args: &GenesisIpcChildGenesisArgs,
+) -> anyhow::Result<()> {
+    // provider with the parent.
+    let parent_provider = IpcProvider::new_with_subnet(
+        None,
+        ipc_provider::config::Subnet {
+            id: args.subnet_id.clone(),
+            config: SubnetConfig::Fevm(EVMSubnet {
+                provider_http: args.parent_endpoint,
+                auth_token: None,
+                registry_addr: args.parent_registry,
+                gateway_addr: args.parent_gateway,
+            }),
+        },
+    )?;
+
+    let genesis_info = parent_provider.get_genesis_info(&args.subnet_id).await?;
+
+    // get gateway genesis
+    let ipc_params = ipc::IpcParams {
+        gateway: ipc::GatewayParams {
+            subnet_id: args.subnet_id.clone(),
+            bottom_up_check_period: genesis_info.bottom_up_checkpoint_period,
+            min_collateral: genesis_info.min_collateral,
+            msg_fee: genesis_info.msg_fee,
+            majority_percentage: genesis_info.majority_percentage,
+            active_validators_limit: genesis_info.active_validators_limit,
+        },
+    };
+    let genesis = Genesis {
+        // We set the genesis epoch as the genesis timestamp so it can be
+        // generated deterministically by all participants
+        // genesis_epoch should be a positive number, we can afford panicking
+        // here if this is not the case.
+        timestamp: Timestamp(genesis_info.genesis_epoch.try_into().unwrap()),
+        chain_name: args.subnet_id.to_string(),
```

Comment:
Shouldn't this be derived from the parent and the address of the subnet actor?

Reply:
It has already been derived in the parent, we just pass the result as an argument here. There are lots of subnet actors deployed in the parent, and we need to signal for which one we want to generate the genesis file for.

Reply:
Sorry I should have been more specific. What I thought was happening is that we have a parent subnet, say, `/r123` and in there we create a subnet which will:
* be an EVM contract with an ID such as `f4101a2b3c...4d5e6f`
* have the subnet ID `/r123/f4101a2b3c...4d5e6f`

Now we want to grab its state to produce a `genesis.json` file for the child subnet via CLI. The set of parameters that minimises any error potential would look something like this:
```
ipc-cli genesis from-parent --parent-url http://mylotus.com:3456 --subnet-actor-addr f4101a2b3c...4d5e6f
```

The `/r123` part of the subnet ID is functionally defined by the `--parent-url`, so all you need is to find the subnet actor, which has a unique ID within this ledger. 

If, on the other hand, we use this:
```
ipc-cli genesis from-parent --parent-url http://mylotus.com:3456 --subnet-id /r123/f4101a2b3c...4d5e6f
```
and then put the subnet ID from the `args` into `genesis.json` then it's a bit more suspicious whether the subnet ID really belongs to the actor. It depends on how it was fetched, ie. whether the full path was taken into account, or just the last leg.

I'm sure it's fine, just wanted to point out that it's a bit over specified.

Reply:
I will not argue against using full subnet IDs whenever they appear, seems like a decent default choice, even though strictly speaking when all operations can only go either to the parent, or to one of the immediate children, an Ethereum address would also do, but at least this way it's easy to tell what it is, and also easy to figure out if it's pointing at the parent or a child. I'll leave UX to others. 

Just have to make sure we always validate that the current subnet is on the path.

Reply:
I see your point, and you are right. UX is something I am not happy about, tbh, and that needs more care. Hopefully we get help for this soon. 

In any case, I will tinker a bit with it, let's see what is more natural.

Reply:
I actually renamed the command to `from-parent`, I think is better. Will push in a minute. I am testing it